make/compiler/Makefile.clang

```diff
@@ -72,8 +72,8 @@ OPT_CFLAGS += $(SANITIZER_OPT_CFLAGS)
 DEF_C_VER := $(shell echo __STDC_VERSION__ | $(CC) -E -x c - | sed -e '/^\#/d' -e 's/L$$//' -e 's/__STDC_VERSION__/0/')
 DEF_CXX_VER := $(shell echo __cplusplus | $(CXX) -E -x c++ - | sed -e '/^\#/d' -e 's/L$$//' -e 's/__cplusplus/0/')
 C_STD := $(shell test $(DEF_C_VER) -lt 199901 && echo -std=gnu99)
-CXX_STD := $(shell test $(DEF_C_VER) -ge 201112 -a $(DEF_CXX_VER) -lt 201103 && echo -std=gnu++11)
-CXX11_STD := $(shell test $(DEF_CXX_VER) -lt 201103 && echo -std=gnu++11)
+CXX_STD := $(shell test $(DEF_C_VER) -ge 201112 -a $(DEF_CXX_VER) -lt 201103 && echo -std=c++11)
+CXX11_STD := $(shell test $(DEF_CXX_VER) -lt 201103 && echo -std=c++11)
```

Comment:
This seems like a potentially out-of-scope change for this PR. Is there a reason we were using `gnu++11` and not `c++11` ? Do these have the same effect on the modern versions? Could you instead change the protobuf plugin to build wit h`-std=gnu++11` ?

Reply:
I believe the protoc compiler itself builds with `c++11` versus `gnu++11` - using `gnu++11` for the plugin did not resolve the issue I was encountering

Reply:
It looks like the change was mostly made to keep the intel compiler consistent with clang and gcc: https://github.com/chapel-lang/chapel/pull/4389#issuecomment-242212484

Reply:
Though there was history of using g99 before that

Reply:
Right but we can pass `-std=c++11` to the protoc compiler without changing it for the rest of Chapel programs.

Reply:
I'm not sure I'm following.  The Chapel protoc plugin relies on the Chapel compiler.  If the compiler is already built, we don't rebuild it with `-std=c++11`, we just build the final plugin-specific portion.  Doing so leads to version incompatibilities because we're building with two different standards.  Are you suggesting that we force building the plugin to recompile the Chapel compiler with that flag?

Reply:
> The Chapel protoc plugin relies on the Chapel compiler. 

How so? The plugin is a C++ project that is loaded by another C++ project. It doesn't include any .chpl code.

Reply:
Ah, looks like you're right.  I got confused because I see the compiler get built during `make protoc-gen-chpl`, but I didn't look as closely at the compilation line for the plugin itself.

That said, I had thought I saw issues that were resolved by remaking the Chapel compiler with this flag instead.  A fresh build with the old state for Makefile.clang today does not seem to show that issue, though, so perhaps I had left a file in a bad state or something?  Weird.  I'll remove that change, thanks.

Reply:
I think that the issue here is that `tools/protoc-gen-chpl/Makefile` is using the compiler configuration for Chapel and the Makefiles that support it (i.e. it does `include $(CHPL_MAKE_HOME)/make/Makefile.base`). My recommendation would be to try to pass `-std=c++11` in the `$(CC) $(CPPFLAGS)` line in `tools/protoc-gen-chpl/Makefile` after the other CPPFLAGS (i.e. in order to override any current setting).

Reply:
For posterity, we discussed that further over chat